### PR TITLE
Recalculate after update missing tracking

### PR DESCRIPTION
**Pull request by NguyenMinhh**
Recalculate tracking count in old flight after remove missing tracking

### CODE REVIEW

**Comment by hoadinh2010** [comment]
image

File: lib/classes/tracking/tracking.js [hunk 1]
```diff
@@ -273,12 +274,32 @@ export default class TrackingWorker {
     });
   }
 
-  findMissingTrackingInUSAHandler(flightId) {
+  findAndUpdateMissingTrackingHandler(flightId) {
     const trackings = this.getListMissingTracking(flightId);
     this.removeListMissingTrackingByListTracking(trackings);
+    this.reCalculateTrackingCountByMissingTrackings(trackings);
```
Comment (hoadinh2010) [inline lib/classes/tracking/tracking.js]: This list trackings are only from one flight, why we must get input the tracking Ids ? Can we use flight ID

Comment (NguyenMinhh) [reply lib/classes/tracking/tracking.js]: After finished add new trackings into new flight, we'll based on list tracking number in current flight, find missing trackings had existed in another flight and remove it.
So this list tracking not only from one flight, it maybe from many flight.

Comment (NguyenMinhh) [reply lib/classes/tracking/tracking.js]: image

File: lib/classes/tracking/tracking.js [hunk 2]
```diff
@@ -274,13 +280,32 @@ export default class TrackingWorker {
...
+  reCalculateTrackingCountByMissingTrackings = (missingTrackings) => {
+    if (missingTrackings.length <= 0) return;
+
+    // Get all flight have missing tracking
+    const lstFlightId = _uniq(missingTrackings.map(missingTracking => missingTracking.flightId), true);
+    const flights = Flights.find({ flightId: { $in: lstFlightId } }).fetch();
+
+    // Update tracking count in each flight
+    flights.forEach((flight) => {
+      const foundTracking = missingTrackings.filter(missingTracking => missingTracking.flightId === flight.flightId).length;
```
Comment (NguyenMinhh) [inline lib/classes/tracking/tracking.js]: I had refactor code when recalculate tracking count, pls check again

**Pull request merged**